openlibrary/core/edits.py

```diff
@@ -286,3 +282,16 @@ def create_comment(cls, username: str, message: str) -> dict[str, str]:
             "message": message,
             # XXX It may be easier to update these comments if they had IDs
         }
+
+    @classmethod
+    def get_counts_by_mode(cls, mode):
+        oldb = db.get_db()
+
+        where_clause = ''
+
+        if mode != 'all':
+            status_list = [f'status={status}' for status in cls.MODES[mode]]
+            where_clause = f'WHERE {" OR ".join(status_list)}'
+
+        query = f"SELECT count(*) from community_edits_queue {where_clause}"
+        return oldb.query(query)[0]['count']
```

Comment:
This function ideally is doing something very similar to `CommunityEditsQueue.get_requests`. The main difference is `get_requests` is returning a (a) a list of objects rather than a count and (b) its limiting the query to a limit.

I wonder if there can be a shared function which both the `CommunityEditsQueue.get_requests` and `get_requests_counts` could use so they both compose the same `where` block but use a different `SELECT` query.

Reply:
Yes! Something like that briefly crossed my mind while making this.  Will add `**kwargs` to `CommunityEditsQueue.get_requests_counts` and create the `WHERE` clause method.  `get_requests_counts` will then be able to handle `reviewer` and `submitter`, and can then be used for roll-ups (or whatever).

Reply:
this solution looks great, I really like your idea of pulling out there `where`. LGTM!